### PR TITLE
Enable SSR for Plugins Detail page

### PR DESCRIPTION
**Pull request by claudiucelfilip**
#### Proposed Changes

More info: p58i-d84-p2#comment-55362
This is a follow-up to #67747  and allows SSR rendering for logged-out Plugin Details pages. 
It contains a part from #66344 (reverted in #67573) and hides changes behind a `plugins/ssr-details` flag.

#### Testing Instructions

* Log out and disable JavaScript
* Go to a few Plugin details pages, ex: 
  * `/plugins/wordpress-seo-premium` (marketplace plugin) 
  * `/plugins/elementor` (org plugin)

* Make sure the plugin details are rendered and (somewhat) match the version with JS enabled
NOTE: Breadcrumbs are not visible and aside content is collapsed 
Screen Shot on 2022-11-02 at 14-05-16

* Log in, enable JS and double-check for regressions in the Plugins section

* To disable the feature flag run: `DISABLE_FEATURES=plugins/ssr-details yarn start`
* Refreshing the `/plugins/wordpress-seo-premium` and `/plugins/elementor` page should  only show the WordPress placeholder logo

Screen Shot on 2022-11-02 at 14-07-52

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

File: client/my-sites/plugins/controller-logged-out.js [hunk 1]
```diff
@@ -139,6 +163,7 @@ export async function fetchCategoryPlugins( context, next ) {
 
 	await prefetchTimebox(
 		[
+			prefetchProductList( queryClient, store ),
```
Comment (claudiucelfilip) [inline client/my-sites/plugins/controller-logged-out.js:166]: Drive-by fix: We still need the Product List to show the price of paid plugins in the category list.

**Review by lsl — APPROVED** [review]
Tests ok, didn't see SSR for logged in, only logged out is that expected?

File: client/my-sites/plugins/index.node.js [hunk 2]
```diff
@@ -4,8 +4,14 @@ import { makeLayout, ssrSetupLocale } from 'calypso/controller';
...
 import { isEnabled } from 'calypso/server/config';
-import { browsePlugins } from './controller';
-import { fetchPlugins, fetchCategoryPlugins, skipIfLoggedIn } from './controller-logged-out';
+import { browsePlugins, browsePluginsOrPlugin } from './controller';
+import {
+	fetchPlugins,
+	fetchCategoryPlugins,
+	fetchPlugin,
+	validatePlugin,
+	skipIfLoggedIn,
```
Comment (lsl) [inline client/my-sites/plugins/index.node.js:13]: :+1:

File: client/my-sites/plugins/index.node.js [hunk 3]
```diff
@@ -38,5 +44,19 @@ export default function ( router ) {
 		);
 	}
 
+	if ( isEnabled( 'plugins/ssr-details' ) ) {
+		router(
+			`/${ langParam }/plugins/:plugin`,
+			skipIfLoggedIn,
```
Comment (lsl) [inline client/my-sites/plugins/index.node.js:50]: :+1:

**Comment by claudiucelfilip** [comment]
@lsl Thanks for the review!
> Tests ok, didn't see SSR for logged in, only logged out is that expected?

Yes, the intention is not to have SSR for logged-in users.
